### PR TITLE
exception_handling/ it-academy.by/ JC1-117-22/Kavaliova Hanna/Done all tasks 

### PR DESCRIPTION
**Pull request by Hanna1201**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

**Review by rakovets — CHANGES_REQUESTED** [review]
You need use logger

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task03Demo.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+public class Task03Demo {
+    public static void main(String[] args) {
+        Task03 task03 = new Task03();
+        String result = "";
+        try {
+            result += task03.adult(17);
+        } catch (AgeException e) {
+            result += e.getMessage();
+            e.printStackTrace();
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task03Demo.java:13]: use logger and stacktrace

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task04.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+import com.rakovets.course.java.core.practice.exception_handling.exeption.NegativeNumberException;
+
+public class Task04 {
+    private int number;
+
+    public int numberIsNegativeException(int number) throws NegativeNumberException {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task04.java:8]:
```suggestion
    public int numberIsNegativeException(int number) {
```

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task03Demo.java (hunk 1 above)
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task03Demo.java:13]: wrong solution

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task04Demo.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+public class Task04Demo {
+    public static void main(String[] args) {
+        Task04 task04 = new Task04();
+        String result = "";
+        try {
+            result += task04.numberIsNegativeException(-2);
+        } catch (NegativeNumberException e) {
+            result += e.getMessage();
+            e.printStackTrace();
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task04Demo.java:13]: wrong solution

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task05.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+import com.rakovets.course.java.core.practice.exception_handling.exeption.DivisionTwoNumbersExcepton;
+
+import java.util.DuplicateFormatFlagsException;
+
+public class Task05 {
+    private int firstNumber;
+    private int secondNumber;
+
+    public int divisionTwoNumber(int firstNumber, int secondNumber) throws DivisionTwoNumbersExcepton {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task05.java:11]:
```suggestion
    public int divisionTwoNumber(int firstNumber, int secondNumber) {
```

**Review by rakovets — CHANGES_REQUESTED** [review]
You need to fix and add correct solutions

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task05.java (hunk 4 above)
Comment (Hanna1201) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task05.java]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task04.java (hunk 2 above)
Comment (Hanna1201) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task04.java]: fixed

**Comment by rakovets** [comment]
outdated

**Pull request closed without merging**